backend/core/src/migration/1588694149283-InitialDB.ts

```diff
@@ -0,0 +1,61 @@
+      undefined
+    )
+    await queryRunner.query(
+      `CREATE TABLE "preference" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "ordinal" character varying, "title" character varying, "subtitle" character varying, "description" character varying, "links" jsonb, "listingId" uuid, CONSTRAINT "PK_5c4cbf49a1e97dcbc695bf462a6" PRIMARY KEY ("id"))`,
+      undefined
+    )
+    await queryRunner.query(
+      `CREATE TABLE "listing" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "acceptingApplicationsAtLeasingAgent" boolean, "acceptingApplicationsByPoBox" boolean, "acceptingOnlineApplications" boolean, "acceptsPostmarkedApplications" boolean, "accessibility" character varying, "amenities" character varying, "applicationDueDate" character varying, "applicationOpenDate" character varying, "applicationFee" character varying, "applicationOrganization" character varying, "applicationAddress" jsonb, "blankPaperApplicationCanBePickedUp" boolean, "buildingAddress" jsonb, "buildingTotalUnits" integer, "buildingSelectionCriteria" character varying, "costsNotIncluded" character varying, "creditHistory" character varying, "criminalBackground" character varying, "depositMin" character varying, "depositMax" character varying, "developer" character varying, "disableUnitsAccordion" boolean, "imageUrl" character varying, "leasingAgentAddress" jsonb, "leasingAgentEmail" character varying, "leasingAgentName" character varying, "leasingAgentOfficeHours" character varying, "leasingAgentPhone" character varying, "leasingAgentTitle" character varying, "name" character varying, "neighborhood" character varying, "petPolicy" character varying, "postmarkedApplicationsReceivedByDate" character varying, "programRules" character varying, "rentalHistory" character varying, "requiredDocuments" character varying, "smokingPolicy" character varying, "unitsAvailable" integer, "unitAmenities" character varying, "waitlistCurrentSize" integer, "waitlistMaxSize" integer, "yearBuilt" integer, CONSTRAINT "PK_381d45ebb8692362c156d6b87d7" PRIMARY KEY ("id"))`,
+      undefined
+    )
+    await queryRunner.query(`CREATE TYPE "attachment_type_enum" AS ENUM('1', '2')`, undefined)
+    await queryRunner.query(
+      `CREATE TABLE "attachment" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "label" character varying, "fileUrl" character varying, "type" "attachment_type_enum", "listingId" uuid, CONSTRAINT "PK_d2a80c3a8d467f08a750ac4b420" PRIMARY KEY ("id"))`,
+      undefined
+    )
+    await queryRunner.query(
+      `ALTER TABLE "unit" ADD CONSTRAINT "FK_72719e0518e726a1de304c6738e" FOREIGN KEY ("listingId") REFERENCES "listing"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`,
```

Comment:
This might not be necessary to think about yet, but do we want all these 1:many listings to live on after their parent listings are removed? I was surprised to see the `on delete no action` here - I'd expect a `cascade` normally?

Reply:
good catch

Reply:
I think `NO ACTION` here actually isn't that bad, since we should more or less never be doing hard deletes from the database, and if it happens by mistake I think we want to limit the damage / preserve the evidence. I realize now that I've said this that we'll inevitably have to write some sort of cleanup script that will be more difficult without cascading deletes, but am I crazy to think that it's worth a little more complexity in that case to be safer otherwise?